lib/scripting/post-process.ts

```diff
@@ -77,19 +194,133 @@ type ConstDeclResult = [ string, null, string, null, Literal ];
+ *          "name":"BallRelease"
+ *       },
+ *       "property":{
+ *          "type":"Identifier",
+ *          "name":"KickBall"
+ *       },
+ *       "computed":false
+ *    },
+ *    null,
+ *    {
+ *       "type":"Literal",
+ *       "value":0
+ *    },
+ *    null,
+ *    [
+ *       {
+ *          "type":"UnaryExpression",
+ *          "operator":"-",
+ *          "prefix":true,
+ *          "argument":{
+ *             "type":"Literal",
+ *             "value":2
+ *          }
+ *       }
+ *    ]
+ * ]
  * ```
  * @todo Literal and UnaryExpression will be more generic in the future!
  */
-export function subCallStmt(result: [ MemberExpression, null, Literal?, null?, UnaryExpression[]? ]) {
+
+export function subCallStmt(result: [MemberExpression, null, Literal?, null?, UnaryExpression[]?]) {
 	const callee = result[0];
-	const firstArg = result[2] ? [ result[2] ] : []; // array, so we can easily spread below
+	const firstArg = result[2] ? [result[2]] : []; // array, so we can easily spread below
 	const otherArgs = result[4] || [];
-	return estree.callExpressionStatement(callee, [ ...firstArg, ...otherArgs] );
+	return estree.callExpressionStatement(callee, [...firstArg, ...otherArgs]);
+}
+
+/**
+ * Grammar:
+ * ```
+ * SubDecl -> "Sub" __ ExtendedID MethodArg:* NL MethodStmt:* _ "End" __ "Sub" NL
+ * ```
+ * Example: `Sub BallRelease_Hit(value1, value2, value3)\n    BallRelease.CreateBall\nEnd Sub\n`
+ * Result:
+ * ```
+ * [
+ *    "Sub",
+ *    null,
+ *    {
+ *       "type":"Identifier",
+ *       "name":"BallRelease_Hit"
+ *    },
+ *    [
+ *       [
+ *          {
+ *             "type":"Identifier",
+ *             "name":"value1"
+ *          },
+ *          [
+ *             {
+ *                "type":"Identifier",
+ *                "name":"value2"
+ *             },
+ *             {
+ *                "type":"Identifier",
+ *                "name":"value3"
+ *             }
+ *          ]
+ *       ]
+ *    ],
+ *    [
+ *       [
+ *          [
+ *             "\n"
+ *          ]
+ *       ]
+ *    ],
+ *    [
+ *       {
+ *          "type":"ExpressionStatement",
+ *          "expression":{
+ *             "type":"CallExpression",
+ *             "callee":{
+ *                "type":"MemberExpression",
+ *                "object":{
+ *                   "type":"Identifier",
+ *                   "name":"BallRelease"
+ *                },
+ *                "property":{
+ *                   "type":"Identifier",
+ *                   "name":"CreateBall"
+ *                },
+ *                "computed":false
+ *             },
+ *             "arguments":[
+ *             ]
+ *          }
+ *       }
+ *    ],
+ *    null,
+ *    "End",
+ *    null,
+ *    "Sub",
+ *    [
+ *       [
+ *          [
+ *             "\n"
+ *          ]
+ *       ]
+ *    ]
+ * ]
+ * ```
+ */
+
+export function subDecl(result: [null, null, Identifier, [[Identifier, Identifier[]]], null, Statement[]?]) {

```

Comment:
This is complicated to understand.

Is it maybe possible to boil those down? I see below there is quite some logic covered, maybe this can be moved to a separate `MethodArg` post-processor that delivers a simpler set of data?